clang/tools/clang-sycl-linker/ClangSYCLLinker.cpp

```diff
@@ -179,60 +182,64 @@ Error executeCommands(StringRef ExecutablePath, ArrayRef<StringRef> Args) {
   return Error::success();
 }
 
-Expected<SmallVector<std::string>> getInput(const ArgList &Args) {
-  // Collect all input bitcode files to be passed to llvm-link.
+/// Returns a new ArgList containg arguments used for the device linking phase.
+DerivedArgList getLinkerArgs(ArrayRef<OffloadFile> Input,
+                             const InputArgList &Args) {
+  DerivedArgList DAL = DerivedArgList(DerivedArgList(Args));
+  for (Arg *A : Args)
+    DAL.append(A);
+  if (DryRun)
+    return DAL;
+
+  // Set the subarchitecture and target triple for this compilation.
+  const OptTable &Tbl = getOptTable();
+  DAL.AddJoinedArg(nullptr, Tbl.getOption(OPT_arch),
+                   Args.MakeArgString(Input.front().getBinary()->getArch()));
+  DAL.AddJoinedArg(nullptr, Tbl.getOption(OPT_triple),
+                   Args.MakeArgString(Input.front().getBinary()->getTriple()));
+
+  return DAL;
+}
+
+/// Collect all fat objects from user input.
+Expected<SmallVector<std::string>> getObjectsFromArgs(const ArgList &Args) {
```

Comment:
```suggestion
SmallVector<std::string> getInputs(const ArgList &Args) {
```
Drop `llvm::Expected` wrapper - function never returns an error.
`getInputs` better reflects the behavior of the function.

Reply:
I am dropping this entire function (getObjectFromArgs) and reverting back to getInputs. 

Thanks